api/src/main/java/net/jqwik/api/ShrinkingDistance.java

```diff
@@ -22,20 +25,44 @@ public static ShrinkingDistance of(long... distances) {
 
 	@API(status = MAINTAINED, since = "1.0")
 	public static <T> ShrinkingDistance forCollection(Collection<Shrinkable<T>> elements) {
-		ShrinkingDistance sumDistanceOfElements = elements
-			.stream()
-			.map(Shrinkable::distance)
-			.reduce(ShrinkingDistance.of(0), ShrinkingDistance::plus);
+		if (elements.isEmpty()) {
+			return MIN;
+		}
 
-		return ShrinkingDistance.of(elements.size()).append(sumDistanceOfElements);
+		long[] data = null;
+		for (Shrinkable<T> element : elements) {
+			long[] next = element.distance().distances;
+			if (data == null) {
+				data = Arrays.copyOf(next, next.length);
+				continue;
+			}
+			data = sumUpArrays(data, data, next);
+
+		}
+		return new ShrinkingDistance(concatArrays(new long[]{elements.size()}, data));
```

Comment:
This could probably be removed by adding `startOffset` argument to `sumUpArrays`.